### ISSUE TITLE
`indented_joins = True` option is not discovered in subdirectories

### ISSUE THREAD

**Issue by pwildenhain** [alanmcruickshank/sqlfluff#255]
## Reprex:
<sup>Windows 10, Python 3.6</sup>
### Set up test project
```
$ mkdir test
$ cd test
$ python -m venv .venv
$ source .venv/Scripts/activate
$ pip install  https://github.com/alanmcruickshank/sqlfluff/archive/master.zip
$ mkdir code
```

#### code/.sqlfluff
```toml
[sqlfluff:rules]
max_line_length = 115

[sqlfluff:indentation]
indented_joins = True
```

#### code/test.sql
```sql
select
    *
from
    my_long_table_name
    inner join my_other_long_table_name on my_other_long_table_name.key = my_long_table_name.key
```

### Linting results

It doesn't respect the `indented_joins` configuration from the parent directory
```
$ sqlfluff lint code
== [code\test.sql] FAIL
L:   5 | P:   5 | L003 | Indentation not consistent with line #3
```

But when we move into the `code` directory, it respects the configuration
```
$ cd code
$ sqlfluff lint
```

### Conclusion

This seems to be isolated to just the `indented_joins = True` option. In the above example, the line length configuration is respected, even from the parent directory.

**Comment by alanmcruickshank** [comment]
good catch, I'm hoping this should be a simple fix.

**Comment by alanmcruickshank** [comment]
@pwildenhain , fyi for debugging this kind of thing in future, the verbose option prints out the config files being found:

```
$ sqlfluff lint code -vv
==== sqlfluff ====
sqlfluff:      0.3.1 python:        3.6.7
dialect:        ansi verbosity:         2
== Raw Config:
core:
    dialect:            ansi
    ignore:             []
    nocolor:            False
    recurse:            True
    templater:          jinja
    verbose:            2
indentation:
    indented_joins:     False
rules:
    indent_unit:        space
    max_line_length:    80
    tab_space_size:     4
    L010:
        capitalisation_policy:consistent
    L019:
        comma_style:        trailing
templater:
    jinja:
        macros:
            dbt_config:         {% macro config() %}{% for k in kwargs %}{% endfor %}{% endmacro %}
            dbt_ref:            {% macro ref(model_ref) %}{{model_ref}}{% endmacro %}
            dbt_source:         {% macro source(source_name, table) %}{{source_name}}_{{table}}{% endmacro %}
            dbt_var:            {% macro var(variable) %}item{% endmacro %}

==== readout ====

=== [ path: code ] ===

== [code\test.sql] PARSING
   Config Diff:
indentation:
    indented_joins:     True
rules:
    max_line_length:    115

== [code\test.sql] FAIL
L:   5 | P:   5 | L003 | Indentation not consistent with line #3
L:   5 | P:  94 | L009 | Files must end with a trailing newline.

==== summary ====
files:             1 violations:        2
clean files:       0 unclean files:     1
avg per file:   2.00 unclean rate:   100%
status:         FAIL
```

Now in this case it looks like it's picking up the config for that folder (see the config diff) section, but for some reason it's not respecting it... 🤔 .

I shall investigate more.

**Comment by alanmcruickshank** [comment]
Yep 🤦 . This is going to be a one-line PR. There's a point in the chain where the config specific for the file wasn't being passed through. Thanks for finding this one!
